crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -1087,15 +1084,16 @@ impl WritebackCache {
     }
 
     // Updates the latest object id cache with an entry that was read from the db.
-    // Writes bypass this function, because an object write is guaranteed to be the
-    // most recent version (and cannot race with any other writes to that object id)
-    //
-    // If there are racing calls to this function, it is guaranteed that after a call
-    // has returned, reads from that thread will not observe a lower version than the
-    // one they inserted
     fn cache_latest_object_by_id(&self, object_id: &ObjectID, object: LatestObjectCacheEntry) {
         trace!("caching object by id: {:?} {:?}", object_id, object);
         self.metrics.record_cache_write("object_by_id");
+        self.insert_cache_mononotic(object_id, object)
+    }
+
+    // Update the object-by-id cache with guaranteed monotonicity. That is, if there are N
+    // calls to the this function from N threads, the write with the highest version will
+    // win the race regardless of what ordering the writes occur in.
+    fn insert_cache_mononotic(&self, object_id: &ObjectID, object: LatestObjectCacheEntry) {
```

Comment:
Maybe mention that caller should usually log and increment `record_cache_write` metric for each call?